app/uk/gov/hmrc/gform/testonly/TestOnlyErrorMessageController.scala

```diff
@@ -0,0 +1,346 @@
+        errorMessageType = reportEn.errorMessageType,
+        debugInfo = reportEn.debugInfo,
+        label_En = reportEn.label,
+        label_Cy = reportCy.label,
+        shortName_En = reportEn.shortName,
+        shortName_Cy = reportCy.shortName,
+        errorShortName_En = reportEn.errorShortName,
+        errorShortName_Cy = reportCy.errorShortName,
+        errorShortNameStart_En = reportEn.errorShortNameStart,
+        errorShortNameStart_Cy = reportCy.errorShortNameStart,
+        errorExample_En = reportEn.errorExample,
+        errorExample_Cy = reportCy.errorExample,
+        errorMessage_En = reportEn.errorMessage,
+        errorMessage_Cy = reportCy.errorMessage,
+        messages_En = reportEn.messages,
+        messages_Cy = reportCy.messages
+      )
+    def htmlEscape(report: FieldErrorAllReport): FieldErrorAllReport =
+      new FieldErrorAllReport(
+        fieldId = HtmlFormat.escape(report.fieldId).toString,
+        errorMessageType = HtmlFormat.escape(report.errorMessageType).toString,
+        debugInfo = HtmlFormat.escape(report.debugInfo).toString,
+        label_En = HtmlFormat.escape(report.label_En).toString,
+        label_Cy = HtmlFormat.escape(report.label_Cy).toString,
+        shortName_En = HtmlFormat.escape(report.shortName_En).toString,
+        shortName_Cy = HtmlFormat.escape(report.shortName_Cy).toString,
+        errorShortName_En = HtmlFormat.escape(report.errorShortName_En).toString,
+        errorShortName_Cy = HtmlFormat.escape(report.errorShortName_Cy).toString,
+        errorShortNameStart_En = HtmlFormat.escape(report.errorShortNameStart_En).toString,
+        errorShortNameStart_Cy = HtmlFormat.escape(report.errorShortNameStart_Cy).toString,
+        errorExample_En = HtmlFormat.escape(report.errorExample_En).toString,
+        errorExample_Cy = HtmlFormat.escape(report.errorExample_Cy).toString,
+        errorMessage_En = HtmlFormat.escape(report.errorMessage_En).toString,
+        errorMessage_Cy = HtmlFormat.escape(report.errorMessage_Cy).toString,
+        messages_En = report.messages_En.map(m => HtmlFormat.escape(m).toString),
+        messages_Cy = report.messages_Cy.map(HtmlFormat.escape(_).toString)
+      )
+    def makeEnCy(reportEn: List[FieldErrorReport], reportCy: List[FieldErrorReport]): List[FieldErrorAllReport] = {
+      val unionList = (reportEn ++ reportCy).groupBy(_.fieldId).map(_._2.head).toList
+
+      unionList.map { element =>
+        val en = reportEn.find(_.fieldId == element.fieldId).getOrElse(FieldErrorReport.makeBlank())
+        val cy = reportCy.find(_.fieldId == element.fieldId).getOrElse(FieldErrorReport.makeBlank())
+        FieldErrorAllReport.make(en, cy)
+      }
+    }
+  }
+
+  def toHtml(report: List[FieldErrorAllReport], formTemplateId: FormTemplateId) = {
+    def reportToRows(report: FieldErrorAllReport): String = {
+      val labelsAndValues = List(
+        ("fieldId", report.fieldId),
+        // ("errorMessageType", report.errorMessageType),
```

Comment:
???

Reply:
to be honest I develop TestOnlyController very quickly, almost add-hoc - just to prove  and test the concept. I need to tidy it up a bit